pkg/systemd/hw-detect.js

```diff
@@ -61,6 +63,60 @@ function findPCI(udevdb, info) {
     }
 }
 
+function findMemoryDevices(udevdb, info) {
+    const memoryArray = [];
+    const dmipath = '/devices/virtual/dmi/id';
+    if (!(dmipath in udevdb)) {
+        return;
+    }
+
+    const props = udevdb[dmipath];
+    // Systemd now exposes memory information in udev, introduced in systemd => 248
+    // https://github.com/systemd/systemd/blob/main/NEWS#L1713
+    if (!('MEMORY_ARRAY_NUM_DEVICES' in props)) {
+        return;
+    }
+
+    for (let slot = 0; slot < props.MEMORY_ARRAY_NUM_DEVICES; slot++) {
```

Comment:
Ugh, is that JS automagically turning a string into a number? (Tested in `node`, seems to work). An explicit `parseInt()` would be a bit cleaner, though.